### PR TITLE
Gutenberg: Add and use jetpack_register_block_type() wrapper

### PR DESCRIPTION
**Pull request by ockham**
This includes a `function_exists()` check to avoid usage of `register_block_type` in older WordPress versions that don't yet have it, leading to fatals (which my recent #11212 caused -- very sorry!)

Some convo (with @dereksmart and @jeherve) at p1549647686423800-slack-jetpack-gutenberg

#### Changes proposed in this Pull Request:

* Add and use `jetpack_register_block_type()` wrapper instead of `register_block_type()`

#### Testing instructions:

- Verify that blocks are still registered and available
- On a test site running WP 4.9.9, verify that no fatals occur.
- Grep for `register_block_type` to make sure we're not calling it directly anymore.

#### Follow-up

As a follow-up, we should modify `tests/php/test_class.jetpack-gutenberg.php` (and `Jetpack_Gutenberg`) so it won't skip tests on a WP installation without `register_block_type` to make sure that `Jetpack_Gutenberg` doesn't fatal.

#### Proposed changelog entry for your changes:

* Add and use `jetpack_register_block_type()` wrapper instead of `register_block_type()`

### CODE REVIEW

File: class.jetpack-gutenberg.php
Comment (enejb) [inline class.jetpack-gutenberg.php]: Lets prefix this with jetpack :)

Comment (jeherve) [reply class.jetpack-gutenberg.php]: Took care of that 👍

File: tests/php/sync/test_class.jetpack-sync-callables.php
Comment (jeherve) [inline tests/php/sync/test_class.jetpack-sync-callables.php]: Could we remove the check here now?

**Review by jeherve — DISMISSED** [review]
This works well for me. I only have one minor comment.

**Comment by ockham** [comment]
Thanks for adding the `jetpack_` prefix, @jeherve -- makes sense to avoid collisions with other plugins that way.

I know I'm the one who suggested the `safe_` infix, but I'm having second thoughts now - `jetpack_safe_register_block_type` is quite a mouthful :sweat_smile: Should we drop the `safe_` after all? (`jetpack_register_block_type` won't collide with the other now-deprecated previous function which was fortunately called `jetpack_register_block`.)

**Comment by jeherve** [comment]
> Should we drop the `safe_` after all?

Yeah, you're right. It seems safe without it. Let's drop it.

File: tests/php/sync/test_class.jetpack-sync-callables.php
Comment (ockham) [reply tests/php/sync/test_class.jetpack-sync-callables.php]: I'll see if that works with WP<5.0 without requiring modifications to the `Jetpack_Gutenberg` class -- otherwise I'll defer to a follow-up PR if that's okay.

Comment (ockham) [reply tests/php/sync/test_class.jetpack-sync-callables.php]: Works: a4e71b18a

**Comment by ockham** [comment]
Changed the function name to `jetpack_register_block_type` and removed the now-obsolete `function_exists()` check from `tests/php/sync/test_class.jetpack-sync-callables.php`.

**Comment by ockham** [comment]
Putting on hold to get #11312 in first.

**Comment by jeherve** [comment]
This should now be good to rebase. 👍

**Comment by ockham** [comment]
Rebased.

**Review by jeherve — APPROVED** [review]
Looking good. Merge when ready.

**Pull request merged**
